csharp/ql/src/utils/modelgenerator/internal/CaptureModels.qll

```diff
@@ -41,18 +41,30 @@ private module Printing implements PrintingSig {
 
 module ModelPrinting = PrintingImpl<Printing>;
 
+/**
+ * Gets the underlying type of the content `c`.
+ */
+private Type getUnderlyingContentType(DataFlow::Content c) {
+  result = getUnderlyingContentTypeSpecific(c) or
+  result = c.(DataFlow::FieldContent).getField().getType() or
+  result = c.(DataFlow::SyntheticFieldContent).getField().getType()
```

Comment:
I think this should be moved into `getUnderlyingContentTypeSpecific` instead (just because Java and C# happen to both have them, other languages might not), which can then be renamed to simply `getUnderlyingContentType`.